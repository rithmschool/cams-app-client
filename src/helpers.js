export const BASE_URL = process.env.REACT_APP_SERVER_URL || 'http://localhost:5000'

export const ensureCorrectUser = (pathId) => {
  let token = localStorage.getItem('token');
  if(!token){return false;}
  let tokenId = JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id
  return +tokenId === +pathId;
}

export const userID = () => JSON.parse(atob(localStorage.getItem('token').split('.')[1])).id

export const config = {
  headers: {
    'Accept':'application/json',
    'ContentType':'application/json',
    'Authorization':'bearer ' + localStorage.getItem('token')
  }
}
