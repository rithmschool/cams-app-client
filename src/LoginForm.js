import React, {PropTypes, Component} from 'react';
import {BASE_URL, config} from './helpers.js';
import axios from 'axios';

class LoginForm extends Component {

  constructor(props) {
    super(props)
    this.state = {
      email: "",
      password: "",
      error: false
    }
  }

  static contextTypes = {
    router: PropTypes.object
  }

  login(config) {
    axios.post(`${BASE_URL}/api/users/auth`, this.state, config)
    .then(response => {
      localStorage.setItem('token', response.data.token)
      this.context.router.history.push('/dashboard')
    })
    .catch(error => {
      this.setState({error: true})
    })
  }

  handleChange(e) {
    this.setState({
      [e.target.name]: e.target.value
    });
  }

  handleSubmit(e) {
    e.preventDefault()
    this.login(config())
  }

  render() {
    let error = (this.state.error) ?
      <p>Invalid credentials</p> : null;
    return (
      <div>
        <div className="banner-text">
          <h1 className="banner-bold">Dashboard</h1>
        </div>
        <div className="content">
        <h1>Login</h1>
        <form onSubmit={this.handleSubmit.bind(this)}>
          <input type="email" name="email" placeholder="email" required onChange={this.handleChange.bind(this)}/>
          <input type="password" name="password" placeholder="password" required onChange={this.handleChange.bind(this)}/>
          <button className="button button-hover" type="submit" value="Submit">Submit</button>
          {error}
        </form>
      </div>
    </div>
    )
  }
}

export default LoginForm;
