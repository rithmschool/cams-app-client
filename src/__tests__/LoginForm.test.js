import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import LoginForm from "../LoginForm";
import toJson from "enzyme-to-json";
import MockAdapter from "axios-mock-adapter";
import { BASE_URL, userID, config } from "../helpers.js";
import axios from "axios";
import LocalStorageMock from "../setupTests/LocalStorageMock";

test("matches snapshot with no errors", () => {
  const login = shallow(<LoginForm />);
  expect(login.text()).not.toContain("Invalid credentials");
  expect(toJson(login)).toMatchSnapshot();
});

test("matches snapshot with errors", () => {
  const login = shallow(<LoginForm />);
  login.setState({ error: true });
  expect(login.text()).toContain("Invalid credentials");
  expect(toJson(login)).toMatchSnapshot();
});

it("should pass a selected value to the onSubmit handler", () => {
  const spy = jest.spyOn(LoginForm.prototype, "handleSubmit");
  const wrapper = shallow(<LoginForm />);
  wrapper.find("form").simulate("submit", { preventDefault() {} });
  expect(spy).toHaveBeenCalled();
});

it("responds to a change", () => {
  const spy = jest.spyOn(LoginForm.prototype, "handleChange");
  const wrapper = shallow(<LoginForm />);
  wrapper.find('input[type="email"]').simulate("change", {
    target: {
      name: "email",
      value: "s"
    }
  });
  expect(spy).toHaveBeenCalled();
});


