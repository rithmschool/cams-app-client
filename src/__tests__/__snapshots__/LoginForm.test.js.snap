// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with errors 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Dashboard
    </h1>
  </div>
  <div
    className="content"
  >
    <h1>
      Login
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
      <button
        className="button button-hover"
        type="submit"
        value="Submit"
      >
        Submit
      </button>
      <p>
        Invalid credentials
      </p>
    </form>
  </div>
</div>
`;

exports[`matches snapshot with no errors 1`] = `
<div>
  <div
    className="banner-text"
  >
    <h1
      className="banner-bold"
    >
      Dashboard
    </h1>
  </div>
  <div
    className="content"
  >
    <h1>
      Login
    </h1>
    <form
      onSubmit={[Function]}
    >
      <input
        name="email"
        onChange={[Function]}
        placeholder="email"
        required={true}
        type="email"
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="password"
        required={true}
        type="password"
      />
      <button
        className="button button-hover"
        type="submit"
        value="Submit"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
